client/client.go

```diff
@@ -247,7 +249,10 @@ func (c *Client) downloadTimestamp() error {
 	raw, err := c.remote.GetMeta(role, maxSize)
 	var s *data.Signed
 	if err != nil || len(raw) == 0 {
-		if err, ok := err.(*store.ErrMetaNotFound); ok {
+		if err, ok := err.(store.ErrMetaNotFound); ok {
+			return err
+		}
+		if old == nil {
 			return err
```

Comment:
This is specific behaviour for the timestamp. If we get an ErrMetaNotFound, we were able to reach the server and it told us it didn't know about the timestamp, so we shouldn't trust whatever we have in cache (because valid timestamps are only generated by the server). The `len(raw) == 0` check is because if the server can return a 200 with an empty body and no content length header, and we will try and parse it.

However, for offline operation, if we can't confirm from the server that we shouldn't trust our local timestamp (i.e. something other than a 404 was returned), we'll use the old one from cache. If `old ==nil` we don't have a valid timestamp in cache. In that instance, we should always defer to the server error because that is the error that must be corrected for the user to be able to succeed in the future.